torch/distributed/_tensor/device_mesh.py

```diff
@@ -101,6 +101,9 @@ def __init__(
         _init_process_groups: bool = True,
     ) -> None:
         self.device_type = device_type
+        self._device_handle = (
```

Comment:
One thing I would suggest is that let's not try to attach `device_handle` to `self`, we need to be a bit careful when adding additional attribute to DeviceMesh, as one thing that is on the radar is that we want to make sure DTensor is pickable, so additional attribute there seems not quite ideal.

We can simply create `_device_handle` and use it in the `__init__` without saving it to device mesh